packages/EasyCore/tests/Doctrine/Dispatchers/DeferredEntityEventDispatcherTest.php

```diff
@@ -0,0 +1,306 @@
+        $newEntity = $this->prophesize(DatabaseEntityInterface::class)->reveal();
+        /** @var \EonX\EasyCore\Interfaces\DatabaseEntityInterface $existedEntity */
+        $existedEntity = $this->prophesize(DatabaseEntityInterface::class)->reveal();
+        $entityCreatedEvent = new EntityCreatedEvent($newEntity);
+        $entityUpdatedEvent = new EntityUpdatedEvent($existedEntity);
+        $eventDispatcher = $this->prophesize(EventDispatcherInterface::class);
+        $eventDispatcher->dispatch($entityCreatedEvent)
+            ->willReturnArgument(0);
+        $eventDispatcher->dispatch($entityUpdatedEvent)
+            ->willReturnArgument(0);
+        /** @var \Symfony\Contracts\EventDispatcher\EventDispatcherInterface $eventDispatcherReveal */
+        $eventDispatcherReveal = $eventDispatcher->reveal();
+        $deferredEntityEventDispatcher = new DeferredEntityEventDispatcher($eventDispatcherReveal);
+
+        $deferredEntityEventDispatcher->deferInsertions([$newEntity], 0);
+        $deferredEntityEventDispatcher->deferUpdates([$existedEntity], 0);
+        $deferredEntityEventDispatcher->dispatch();
+
+        $eventDispatcher->dispatch($entityCreatedEvent)
+            ->shouldHaveBeenCalledOnce();
+        $eventDispatcher->dispatch($entityUpdatedEvent)
+            ->shouldHaveBeenCalledOnce();
+        self::assertSame([], $this->getPrivatePropertyValue($deferredEntityEventDispatcher, 'entityInsertions'));
+        self::assertSame([], $this->getPrivatePropertyValue($deferredEntityEventDispatcher, 'entityUpdates'));
+    }
+
+    public function testDispatchSucceedsWithoutEntities(): void
+    {
+        $eventDispatcher = $this->prophesize(EventDispatcherInterface::class);
+        /** @var \Symfony\Contracts\EventDispatcher\EventDispatcherInterface $eventDispatcherReveal */
+        $eventDispatcherReveal = $eventDispatcher->reveal();
+        $deferredEntityEventDispatcher = new DeferredEntityEventDispatcher($eventDispatcherReveal);
+
+        $deferredEntityEventDispatcher->dispatch();
+
+        $eventDispatcher->dispatch(Argument::type(EntityCreatedEvent::class))
+            ->shouldNotHaveBeenCalled();
+        $eventDispatcher->dispatch(Argument::type(EntityUpdatedEvent::class))
+            ->shouldNotHaveBeenCalled();
+        self::assertSame([], $this->getPrivatePropertyValue($deferredEntityEventDispatcher, 'entityInsertions'));
+        self::assertSame([], $this->getPrivatePropertyValue($deferredEntityEventDispatcher, 'entityUpdates'));
+    }
+
+    public function testEnableSucceeds(): void
+    {
+        /** @var \Symfony\Contracts\EventDispatcher\EventDispatcherInterface $eventDispatcherReveal */
+        $eventDispatcherReveal = $this->prophesize(EventDispatcherInterface::class)->reveal();
+        $deferredEntityEventDispatcher = new DeferredEntityEventDispatcher($eventDispatcherReveal);
+
+        $deferredEntityEventDispatcher->disable();
```

Comment:
Let's move this line to the arrange block.

Reply:
Done